script.js

```diff
@@ -1,4 +1,190 @@
+
+    console.log(
+      playerTwoScore
+        .map(function (elt) {
+          // assure the value can be converted into an integer
+          return /^\d+$/.test(elt) ? parseInt(elt) : 0;
+        })
+        .reduce(function (a, b) {
+          // sum all resulting numbers
+          return a + b;
+        })
+    );
+  }
+  return `Player 1 total score is ${playerOneTotalScore} and Player 2 total score is ${playerTwoTotalScore}.`;
+};
+
+// roll dice to string
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber.toString();
+};
+
+// player 1 dice roll
+var playerOneDiceRoll = function () {
+  playerOneDiceOne = rollDice();
+  playerOneDiceTwo = rollDice();
+  gameMode = "player 1 position";
+  return `Hello Player 1 🐶.<br><br>You rolled ${playerOneDiceOne} for Dice 🎲 One and ${playerOneDiceTwo} for Dice 🎲 Two.<br><br>Now please choose the order of your Dice 🎲 by entering 1 or 2.`;
+};
+
+// player 1 choose dice position
+var playerOneChoice = function (input) {
+  if (input == 1 || input == 2) {
+    playerOneNumber = combineDiceRolls(input);
+    playerOneScore.push(playerOneNumber);
+    gameMode = "player 2";
+    return `Player 1 🐶.<br><br>Your chosen number is ${playerOneNumber}.<br><br>It is now Player 2's 🐹 turn to give it a go.<br><br>Click submit to roll the dice 🎲.`;
+  } else {
+    return verifyInput(input);
+  }
+};
+
+// player 2 dice roll
+var playerTwoDiceRoll = function () {
+  playerTwoDiceOne = rollDice();
+  playerTwoDiceTwo = rollDice();
+  gameMode = "player 2 position";
+  return `Hello Player 2 🐹.<br><br>You rolled ${playerTwoDiceOne} for Dice 🎲 One and ${playerTwoDiceTwo} for Dice 🎲 Two.<br><br>Now please choose the order of your Dice 🎲 by entering 1 or 2.`;
+};
+
+// player 2 choose dice position
+var playerTwoChoice = function (input) {
+  if (input == 1 || input == 2) {
+    playerTwoNumber = combineDiceRolls(input);
+    playerTwoScore.push(playerTwoNumber);
+    gameMode = "score time";
+    return `Player 2 🐹.<br><br>Your chosen number is ${playerTwoNumber}.<br><br>Let's see who's the winner?! 🍾`;
+  } else {
+    return verifyInput(input);
+  }
+};
+
+// verify if input is a valid number
+var verifyInput = function (input) {
+  if (input != 1 || input != 2) {
+    return "🙅 Please enter 1 or 2 to position your dice 🎲 roll.";
+  }
+};
+
+// combine the dice rolls
+var combineDiceRolls = function (input) {
+  if (input == 1 && gameMode == "player 1 position") {
+    return playerOneDiceOne + playerOneDiceTwo;
+  } else if (input == 1 && gameMode == "player 2 position") {
+    return playerTwoDiceOne + playerTwoDiceTwo;
+  } else if (input == 2 && gameMode == "player 1 position") {
+    return playerOneDiceTwo + playerOneDiceOne;
+  } else if (input == 2 && gameMode == "player 2 position") {
+    return playerTwoDiceTwo + playerTwoDiceOne;
+  }
+};
```

Comment:
Good that you have a function just to handle combination of the dice rolls. it can help with the scalability of the code if you want to work on it in the future but the way the combination is done is not scalable maybe you can explore how to make this part work no matter how many players you have as an exercise. 

another comment is to put helper functions above other functions that are calling it for readability